### DISCUSSION TITLE
Add support for AzurePipelines StartGroup and EndGroup commands similar to the GitHubActions commands

### DISCUSSION THREAD

**Discussion by arturcic** [cake-build/cake#4103] [Ideas]
Here are the docs:

https://learn.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#formatting-commands.

**Comment by JuergenRB** [comment]
Have you seen the project [Cake.BuildSystems.Module](https://github.com/cake-contrib/Cake.BuildSystems.Module) ? It supports the representation of groups in Azure Pipelines

**Reply by arturcic** [reply]
I have not. I will check it out, thanks

**Reply by arturcic** [reply]
I had a look, and it seems the [WriteGroupCommand](https://github.com/cake-contrib/Cake.BuildSystems.Module/blob/d0b9909f31fb6f6ae984d894e7042aebb695266b/src/Cake.AzurePipelines.Module/AzurePipelinesEngine.cs#L153) is private, meaning it cannot be accessible in the code if I want to create my own group (a nested group inside of the Task group), something I can achieve if I use the GitHubActions.Commands

**Reply by JuergenRB** [reply]
Unfortunately it won't work as neither [GitHub Actions](https://github.com/actions/runner/issues/802) nor AzurePipelines support nested groups.
For AzurePipelines I have not found any information that this is planned.

If you only want to create your own groups, you could use the normal [Console.Output](https://cakebuild.net/docs/writing-builds/console-output) in AzurePipelines
```csharp
Task("Test")
    .Does(() =>
{
    Information("##[group]Beginning of a group");
    Information("##[warning]Warning message");
    Information( "##[error]Error message");
    Information("##[section]Start of a section");
    Information( "##[debug]Debug text");
    Information( "##[command]Command-line being run");
    Information( "##[endgroup]");
});

```
![image](https://user-images.githubusercontent.com/22403655/208391068-546a84db-226e-4fcc-9526-37201a8f6a03.png)
